File: URI/1 - INICIANTE/Python/1929 - Triangulo.py
a, b, c, d = map(int, input().split())
if abs(b - c) < a and a < b + c and abs(a - c) < b and b < a + c and abs(b - a) < c and c < b + a:
    print('S')
elif abs(b - c) < d and d < b + c and abs(d - c) < b and b < d + c and abs(b - d) < c and c < b + d:
    print('S')
elif abs(d - c) < a and a < d + c and abs(a - c) < d and d < a + c and abs(d - a) < c and c < d + a:
    print('S')
elif abs(b - d) < a and a < b + d and abs(a - d) < b and b < a + d and abs(b - a) < d and d < b + a:
    print('S')
else:
    print('N')

File: URI/1 - INICIANTE/C/1929 - Triangulo.c
#include <stdio.h>
#include <math.h>

int main(){
	int a, b, c, d;
	scanf(" %d %d %d %d", &a, &b, &c, &d);
	if(abs(b - c) < a && a < b + c && abs(a - c) < b && b < a + c && abs(b - a) < c && c < b + a){
		printf("S\n");
	}else if(abs(b - c) < d && d < b + c && abs(d - c) < b && b < d + c && abs(b - d) < c && c < b + d){
		printf("S\n");
	}else if(abs(d - c) < a && a < d + c && abs(a - c) < d && d < a + c && abs(d - a) < c && c < d + a){
		printf("S\n");
	}else if(abs(b - d) < a && a < b + d && abs(a - d) < b && b < a + d && abs(b - a) < d && d < b + a){
		printf("S\n");
	}else{
		printf("N\n");
	}
	return 0;
}
